identity/src/test/java/com/android/identity/UtilTest.java

```diff
@@ -783,6 +783,49 @@ public void fromHexThrows() {
         assertThrows(IllegalArgumentException.class, () -> Util.fromHex("XX"));
     }
 
+    @Test
+    public void asString() {
+        final DataItem dataItem = new co.nstant.in.cbor.model.UnicodeString("hello");
+        assertEquals( "hello", Util.asString(dataItem));
+    }
+
+    @Test
+    public void asStringThrows() {
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(co.nstant.in.cbor.model.SimpleValue.TRUE));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(new co.nstant.in.cbor.model.UnsignedInteger(42)));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asString(new co.nstant.in.cbor.model.ByteString(new byte[] { 1, 2, 3 })));
+    }
+
+    @Test
+    public void asLong() {
+        final DataItem dataItem = new co.nstant.in.cbor.model.UnsignedInteger(8675309);
+        assertEquals( 8675309, Util.asLong(dataItem));
+    }
+
+    @Test
+    public void asLongThrowsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asLong(co.nstant.in.cbor.model.SimpleValue.TRUE));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asLong(new co.nstant.in.cbor.model.UnicodeString("not a number")));
+        assertThrows(IllegalArgumentException.class,
+            () -> Util.asLong(new co.nstant.in.cbor.model.ByteString(new byte[] { 1, 2, 3 })));
+    }
+
+    @Test
+    public void asLongThrowsArithmeticException() {
+        // One bit too big: 0001 0000 0000 0000 0000
+        final BigInteger tooBig = new BigInteger(new byte[] {
```

Comment:
I think the following is easier to read/verify (I don't want to be counting bytes):

   BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE)

We also want to check all boundary cases, not just this one.

// Should not throw but return the right value:
BigInteger.valueOf(Long.MIN_VALUE)
BigInteger.valueOf(Long.MAX_VALUE)

// Should throw:
BigInteger.valueOf(Long.MIN_VALUE).subtract(BigInteger.ONE)
BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE)

?